src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -429,8 +471,15 @@ public static GapicProductConfig createDummyInstance(
   private static ImmutableMap<String, SingleResourceNameConfig> createSingleResourceNameConfigs(
       DiagCollector diagCollector,
       ConfigProto configProto,
-      ProtoFile file,
-      TargetLanguage language) {
+      @Nullable List<ProtoFile> sourceProtos,
+      TargetLanguage language,
+      ProtoParser protoParser) {
+    ProtoFile file = null;
+    if (sourceProtos == null) {
+      sourceProtos = new LinkedList<>();
```

Comment:
`ImmutableList.of()`? I don't think we're mutating this list right?

Reply:
Done.